### PR TITLE
Created Identifier to help keep track of plugin resources. 

### PR DESCRIPTION
**Pull request by neeleshpoli**
Created Identifier to help keep track of plugin resources. 
### Use cases: 
- Can be used in the event handler api to make it easy to keep track of events
- Make it easier to keep track of commands.
- \+ many other uses as well

Serde was added so that structs can be serialized and deserialized when transporting between plugin and server. Currently message pack is used, but protocol buffers should be used in the future due to smaller size and efficiency. The only reason why protocol buffers are not used right now is because it requires some work to setup.

### CODE REVIEW

File: pumpkin-plugin/Cargo.toml [hunk 1]
```diff
@@ -5,4 +5,5 @@ edition.workspace = true
 
 [dependencies]
 extism = "1.5.0"
-log.workspace = true
\ No newline at end of file
+log.workspace = true
+serde ={ version = "1.0.209", features = ["derive"] }
```
Comment (Snowiiii) [inline pumpkin-plugin/Cargo.toml:9]: Hey can you may just use serde version `1.0`. I really should move serde into workspace

Comment (neeleshpoli) [reply pumpkin-plugin/Cargo.toml:9]: Sure!

Comment (neeleshpoli) [reply pumpkin-plugin/Cargo.toml:9]: I can move it into the workspace if you're fine with it

**Comment by neeleshpoli** [comment]
I have moved serde into the workspace as well

**Comment by Snowiiii** [comment]
Okay looks good then, Thank you @neeleshpoli

**Pull request merged**